mirascope/beta/openai/parse.py

```diff
@@ -21,7 +21,7 @@ def parse(
     model: Literal["gpt-4o-2024-08-06"],
     response_model: type[_ResponseModelT],
     call_params: OpenAICallParams | None = None,
-):
+) -> Callable[_P, _ResponseModelT | Awaitable[_ResponseModelT]]:
```

Comment:
this is purposefully untyped in order to take advantage of the overloads defined internally. If we want to type this, I believe we'll need to use a protocol